vtk build failure when VCPKG_BUILD_TYPE=release

**Host Environment**
- OS: Windows 10
- Compiler: VS2019

**To Reproduce**
Steps to reproduce the behavior:
In PowerShell:
```
cp .\triplets\x64-windows.cmake .\triplets\x64-windows-release.cmake
Add-Content .\triplets\x64-windows-release.cmake "set(VCPKG_BUILD_TYPE release)"
./vcpkg install vtk
```

**Failure logs** 
```
CMake Error at ports/vtk/portfile.cmake:201 (file):
  file failed to open for reading (No such file or directory):

    C:/Users/Ross Taylor/Q/vcpkg/packages/vtk_x64-windows-release/share/vtk/VTK-targets-debug.cmake
```

**Additional context**
If only release is built there is no VTK-targets-debug.cmake file generated so the code that tries to modify it fails. A simple fix is to verify the file is there (in portfile.cmake line 199):
```
    # we also have to bend the lines referencing the tools in VTKTargets-debug.cmake
    # to make them point to the release version of the tools
    set(filename ${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake)
    if(EXISTS ${filename})
        file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" VTK_TARGETS_CONTENT_DEBUG)
        string(REPLACE "debug/bin/${TOOL_NAME}" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_DEBUG "${VTK_TARGETS_CONTENT_DEBUG}")
        string(REPLACE "tools/vtk/${TOOL_NAME}d" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_DEBUG "${VTK_TARGETS_CONTENT_DEBUG}")
        file(WRITE "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" "${VTK_TARGETS_CONTENT_DEBUG}")
    endif()
```

--- a/ports/vtk/CONTROL
+++ b/ports/vtk/CONTROL
@@ -1,5 +1,5 @@
 Source: vtk
-Version: 9.0
+Version: 9.0-1
 Description: Software system for 3D computer graphics, image processing, and visualization
 Homepage: https://github.com/Kitware/VTK
 Build-Depends: zlib, libpng, tiff, libxml2, jsoncpp, glew, freetype, expat, hdf5[core], libjpeg-turbo, proj4, lz4, liblzma, libtheora, eigen3, double-conversion, pugixml, libharu[notiffsymbols], sqlite3, netcdf-c, utfcpp, libogg, pegtl-2

--- a/ports/vtk/portfile.cmake
+++ b/ports/vtk/portfile.cmake
@@ -196,7 +196,7 @@ function(_vtk_remove_debug_tool TOOL_NAME)
     if(EXISTS ${filename})
         file(REMOVE ${filename})
     endif()
-    if (NOT CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL debug)
+    if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL debug)
         # we also have to bend the lines referencing the tools in VTKTargets-debug.cmake
         # to make them point to the release version of the tools
         file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" VTK_TARGETS_CONTENT_DEBUG)
@@ -214,7 +214,7 @@ function(_vtk_move_release_tool TOOL_NAME)
         file(REMOVE ${old_filename})
     endif()
 
-    if (NOT CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL release)
+    if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL release)
         # we also have to bend the lines referencing the tools in VTKTargets-release.cmake
         # to make them point to the tool folder
         file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-release.cmake" VTK_TARGETS_CONTENT_RELEASE)

--- a/ports/vtk/portfile.cmake
+++ b/ports/vtk/portfile.cmake
@@ -196,12 +196,14 @@ function(_vtk_remove_debug_tool TOOL_NAME)
     if(EXISTS ${filename})
         file(REMOVE ${filename})
     endif()
-    # we also have to bend the lines referencing the tools in VTKTargets-debug.cmake
-    # to make them point to the release version of the tools
-    file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" VTK_TARGETS_CONTENT_DEBUG)
-    string(REPLACE "debug/bin/${TOOL_NAME}" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_DEBUG "${VTK_TARGETS_CONTENT_DEBUG}")
-    string(REPLACE "tools/vtk/${TOOL_NAME}d" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_DEBUG "${VTK_TARGETS_CONTENT_DEBUG}")
-    file(WRITE "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" "${VTK_TARGETS_CONTENT_DEBUG}")
+    if (NOT CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL debug)
+        # we also have to bend the lines referencing the tools in VTKTargets-debug.cmake
+        # to make them point to the release version of the tools
+        file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" VTK_TARGETS_CONTENT_DEBUG)
+        string(REPLACE "debug/bin/${TOOL_NAME}" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_DEBUG "${VTK_TARGETS_CONTENT_DEBUG}")
+        string(REPLACE "tools/vtk/${TOOL_NAME}d" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_DEBUG "${VTK_TARGETS_CONTENT_DEBUG}")
+        file(WRITE "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-debug.cmake" "${VTK_TARGETS_CONTENT_DEBUG}")
+    endif()
 endfunction()
 
 # Move the release binary TOOL_NAME from bin to tools
@@ -212,11 +214,13 @@ function(_vtk_move_release_tool TOOL_NAME)
         file(REMOVE ${old_filename})
     endif()
 
-    # we also have to bend the lines referencing the tools in VTKTargets-release.cmake
-    # to make them point to the tool folder
-    file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-release.cmake" VTK_TARGETS_CONTENT_RELEASE)
-    string(REPLACE "bin/${TOOL_NAME}" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_RELEASE "${VTK_TARGETS_CONTENT_RELEASE}")
-    file(WRITE "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-release.cmake" "${VTK_TARGETS_CONTENT_RELEASE}")
+    if (NOT CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL release)
+        # we also have to bend the lines referencing the tools in VTKTargets-release.cmake
+        # to make them point to the tool folder
+        file(READ "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-release.cmake" VTK_TARGETS_CONTENT_RELEASE)
+        string(REPLACE "bin/${TOOL_NAME}" "tools/vtk/${TOOL_NAME}" VTK_TARGETS_CONTENT_RELEASE "${VTK_TARGETS_CONTENT_RELEASE}")
+        file(WRITE "${CURRENT_PACKAGES_DIR}/share/vtk/VTK-targets-release.cmake" "${VTK_TARGETS_CONTENT_RELEASE}")
+    endif()
 endfunction()
 
 set(VTK_SHORT_VERSION 9.0)

